### PR TITLE
feat(ui): demo warning banner and input amount limits

### PR DESCRIPTION
**Pull request by th0rOdinson**
## Summary

- Adds a warning banner to the cross-chain page ("This is a demo app for testing and experimentation. Do not use large amounts.")
- Caps input amounts per token symbol: USDC/USDT/DAI at 100, ETH/WETH at 0.03, BTC/WBTC at 0.001
- Submit button shows "Amount too large for demo" and disables when the limit is exceeded
- Tokens not in the limit map are unrestricted

## Test plan

- 21 unit tests for `exceedsDemoLimit` covering all token types, edge cases (undefined/empty symbol, invalid amounts, case insensitivity, comma decimals)
- 4 Playwright e2e tests in `demo-limits.spec.ts` (banner visibility, submit disabled at 101 USDC, enabled at 100, disabled in build quote mode)

Closes EFI-865

### CODE REVIEW

File: examples/ui/app/cross-chain/components/SwapForm.tsx [hunk 1]
```diff
@@ -228,6 +244,7 @@ export function SwapForm({ onSubmit, onInputChange, isLoading = false, isDisable
     if (!isConnected) return 'Connect Wallet';
     if (isLoading) return mode === 'buildQuote' ? 'Building Quote...' : 'Fetching Quotes...';
     if (hasInsufficientBalance) return 'Insufficient Balance';
+    if (exceedsLimit) return 'Amount too large for demo';
     return mode === 'buildQuote' ? 'Build Quote' : 'Get Quotes';
```
[the remark this answers is not in the document]
Comment (th0rOdinson) [reply examples/ui/app/cross-chain/components/SwapForm.tsx:248]: done!

File: examples/ui/app/cross-chain/utils/amountValidation.ts [hunk 2]
```diff
@@ -27,6 +29,16 @@ export function isValidAmount(value: string): boolean {
   return !isNaN(parseFloat(normalized));
 }
 
+/** Returns true if the amount exceeds the demo limit for the given token symbol. */
+export function exceedsDemoLimit(amount: string, symbol: string | undefined): boolean {
+  if (!symbol?.trim()) return false;
+  const key = symbol.trim();
+  if (!(key in DemoToken)) return false;
+  const max = DEMO_MAX_AMOUNT[key as DemoToken];
```
[the remark this answers is not in the document]
Comment (th0rOdinson) [reply examples/ui/app/cross-chain/utils/amountValidation.ts:37]: Not an issue. The symbol comes from `inputTokenInfo.symbol` which is set by the provider with canonical casing (e.g. `USDC`, `cbBTC`). There is no user-facing path where the symbol could arrive in a different case. The whitelist in `routeSelection.ts` already filters by exact match (`WHITELISTED_SYMBOLS.has(token.symbol)`), so a token with unexpected casing would not appear in the UI at all.

**Comment by th0rOdinson** [comment]
@cubic-dev-ai approve

**Review by 0xPhantomm — APPROVED** [review]
